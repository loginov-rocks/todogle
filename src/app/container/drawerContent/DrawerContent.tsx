import {
  CircularProgress, Divider, List, ListItem, ListItemIcon, ListItemText,
  ListSubheader, makeStyles,
} from '@material-ui/core';
import { Add, ExitToApp, Inbox } from '@material-ui/icons';
import * as React from 'react';
import { useSelector } from 'react-redux';
import { useHistory } from 'react-router-dom';

import * as R from '../../../routes';
import gapi from '../../../services/gapi';
import { getTaskListsLoaded } from '../../../store/selectors';

import TaskLists from './taskLists/TaskLists';

const useStyles = makeStyles(theme => ({
  toolbarMock: theme.mixins.toolbar,
}));

interface Props {
  onMenuItemClick?: () => void;
}

const DrawerContent = ({ onMenuItemClick }: Props) => {
  const history = useHistory();
  const taskListsLoaded = useSelector(getTaskListsLoaded);
  const dynamicStyles = useStyles();

  const handleHomeClick = () => {
    history.push(R.HOME);
    if (onMenuItemClick) {
      onMenuItemClick();
    }
  };

  const handleCreateTaskListClick = () => {
    history.push(R.CREATE_TASK_LIST);
    if (onMenuItemClick) {
      onMenuItemClick();
    }
  };

  const handleSignOutClick = () => {
    gapi.signOut();
    if (onMenuItemClick) {
      onMenuItemClick();
    }
  };

  return (
    <>

      <div className={dynamicStyles.toolbarMock} />

      <Divider />

      <List dense>
        <ListItem button onClick={handleHomeClick}>
          <ListItemIcon><Inbox /></ListItemIcon>
          <ListItemText primary="Inbox" />
        </ListItem>
      </List>

      <Divider />

      {taskListsLoaded ? (
        <>
          <List dense subheader={<ListSubheader>Task Lists</ListSubheader>}>
            <ListItem button onClick={handleCreateTaskListClick}>
              <ListItemIcon><Add /></ListItemIcon>
              <ListItemText primary="Create New" />
            </ListItem>
            <TaskLists onTaskListClick={onMenuItemClick} />
          </List>
        </>
      ) : (
        <CircularProgress />
      )}

      <Divider />

      <List dense>
        <ListItem button onClick={handleSignOutClick}>
          <ListItemIcon><ExitToApp /></ListItemIcon>
          <ListItemText primary="Sign Out" />
        </ListItem>
      </List>

    </>
  );
};

export default DrawerContent;
