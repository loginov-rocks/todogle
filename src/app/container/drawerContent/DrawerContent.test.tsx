import { shallow } from 'enzyme';
import * as React from 'react';

import DrawerContent from './DrawerContent';

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
}));

it('matches snapshot', () => {
  const wrapper = shallow(
    <DrawerContent />,
  );

  expect(wrapper).toMatchSnapshot();
});
