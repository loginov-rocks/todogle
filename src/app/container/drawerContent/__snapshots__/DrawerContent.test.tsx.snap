// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Fragment>
  <div
    className="makeStyles-toolbarMock-1"
  />
  <WithStyles(ForwardRef(Divider)) />
  <WithStyles(ForwardRef(List))
    dense={true}
  >
    <WithStyles(ForwardRef(ListItem))
      button={true}
      onClick={[Function]}
    >
      <WithStyles(ForwardRef(ListItemIcon))>
        <InboxIcon />
      </WithStyles(ForwardRef(ListItemIcon))>
      <WithStyles(ForwardRef(ListItemText))
        primary="Inbox"
      />
    </WithStyles(ForwardRef(ListItem))>
  </WithStyles(ForwardRef(List))>
  <WithStyles(ForwardRef(Divider)) />
  <WithStyles(ForwardRef(CircularProgress)) />
  <WithStyles(ForwardRef(Divider)) />
  <WithStyles(ForwardRef(List))
    dense={true}
  >
    <WithStyles(ForwardRef(ListItem))
      button={true}
      onClick={[Function]}
    >
      <WithStyles(ForwardRef(ListItemIcon))>
        <ExitToAppIcon />
      </WithStyles(ForwardRef(ListItemIcon))>
      <WithStyles(ForwardRef(ListItemText))
        primary="Sign Out"
      />
    </WithStyles(ForwardRef(ListItem))>
  </WithStyles(ForwardRef(List))>
</Fragment>
`;
